core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java

```diff
@@ -129,14 +137,28 @@ private int expectedScanTaskCount(int partitions) {
         }
       case DATA_FILES:
       case DELETE_FILES:
+      case ALL_DELETE_FILES:
         return partitions;
       case ALL_DATA_FILES:
         return partitions * 2; // ScanTask for Data Manifest in DELETED and ADDED states
+      case ALL_FILES:
+        if (formatVersion == 1) {
+          return partitions * 2; // ScanTask for Data Manifest in DELETED and ADDED states
+        } else {
+          return partitions * 4; // ScanTask for Delete and Data File in DELETED and ADDED states
+        }
       default:
         throw new IllegalArgumentException("Unsupported metadata table type:" + type);
     }
   }
 
+  private boolean allFileTableTest(MetadataTableType tableType) {
```

Comment:
Nit: This might read better as `isAllFilesTable`, though I recognize there is an `AllFilesTable` now.

Maybe `isOneOfAllFilesType` or `isTableAggregatedFromAllSnapshots`?

Given there will be an `ALL_FILES` table, I'm not sure of the best name to add for this but throwing those out there.

Reply:
What about `isAggFileTable`?